### PR TITLE
GEOPY-290: Update grav/mag inversion app 

### PR DESCRIPTION
**Pull request by domfournier**
**GEOPY-290 - Update inversion app (ui.json  + SIMPEG)** 
 - Make the app run from the ui.json with params/validators.
- Adresses #9

### CODE REVIEW

**Review by benk-mira — DISMISSED** [review]
Nice work here.  Can't wait to get this merged in with the DC stuff!  Made a few comments - none are blocking.  I'll let you use your discretion for what to fix up.

File: geoapps/drivers/base_inversion.py [hunk 1]
```diff
@@ -40,8 +41,7 @@
     InversionWindow,
 )
 
-cluster = LocalCluster(processes=False)
-client = Client(cluster)
+warnings.filterwarnings("ignore")
```
[the remark this answers is not in the document]
Comment (domfournier) [reply geoapps/drivers/base_inversion.py:44]: That's to get rid of the skleanr kmeans warning

File: geoapps/drivers/base_inversion.py [hunk 2]
```diff
@@ -262,24 +281,6 @@ def run(self):
         self.save_residuals(self.inversion_data.data_entity, dpred)
         self.finish_inversion_message(dpred)
 
-    def get_weighting_matrix(self, global_misfit):
```
[the remark this answers is not in the document]
Comment (domfournier) [reply geoapps/drivers/base_inversion.py:265]: agreed

[the remark this answers is not in the document]
Comment (benk-mira) [reply geoapps/drivers/base_inversion.py:265]: yes - by update sensitivity matrix

File: geoapps/base.py [hunk 3]
```diff
@@ -90,14 +89,18 @@ def __init__(self, **kwargs):
         self.ga_group_name.observe(self.ga_group_name_update)
         self.__populate__(**self.defaults)
 
+        for value in self.__dict__.copy():
```
[the remark this answers is not in the document]
Comment (domfournier) [reply geoapps/base.py:92]: Good catch. Copy wasn't necessary at all. Switch to `for key in self.__dict__:`

**Comment by benk-mira** [comment]
Laundry list from testing ui:
- w_cell_size not updating
- z_from_topo switch not updating
- forward bool should write a forward ui.json
- constant offsets not updating
- gravity defaults to Gxz channel

**Comment by benk-mira** [comment]
Nice work on the laundry list.  I've opened an issue to fix the detrend order: GEOPY-327 as it's not critical to this issue.

**Review by benk-mira — APPROVED** [review]
Great Work Dom - that's a big one!

**Pull request merged**